CefSharp/Internals/JavascriptObjectRepository.cs

```diff
@@ -314,6 +317,14 @@ internal bool TryCallMethod(long objectId, string name, object[] parameters, out
                     }
                     else
                     {
+                        // anyone extending the Method Interceptor should be able to use Task 
+                        if (obj.MethodInterceptor.IsAsynchronous)
+                        {
+                            // run the asynchronous interceptor without worrying about creating a deadlock 
```

Comment:
Because we can't use an actual `await` here without changing the entire method body, most people opt to do `.GetAwaiter().GetResult()`, however this has been known to cause deadlocks in a one-thread-at-a-time context, and because we don't know what the implementing function's Task look like, it's safer to spawn on a new Task and call `.Result`

I should say in our fork, interceptor is still synchronous, and I made this change mostly as a QOL thing.

Reply:
At a minimum we can extract this into it's own class without changing the current interceptor implementation see https://github.com/amaitland/CefSharp/commit/3bd359d37e88c5d52d82cf2e001596e9c2d536c9#diff-9ae6db7388c094d4a39bd83c3885526bR14 for an example. 

I'd like to see this handled asynchronously,  I'll wait until we have some unit test coverage then have a go at rewriting.